pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -284,10 +284,22 @@ func validateStateChangeRequests(ar *v1beta1.AdmissionRequest, vm *v1.VirtualMac
 		}
 
 		if getRenameRequest(existingVM) != nil {
-			return []metav1.StatusCause{{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: "Modifying a VM during a rename process is not allowed",
-			}}
+			allowed := webhooks.GetAllowedServiceAccounts()
+			if _, ok := allowed[ar.UserInfo.Username]; ok {
+				if !reflect.DeepEqual(existingVM.Spec, vm.Spec) {
+					return []metav1.StatusCause{{
+						Type:    metav1.CauseTypeFieldValueNotSupported,
+						Message: fmt.Sprint("Cannot update VM spec until rename process completes"),
+						Field:   k8sfield.NewPath("spec").String(),
+					}}
+
+				}
+			} else {
+				return []metav1.StatusCause{{
+					Type:    metav1.CauseTypeFieldValueNotSupported,
+					Message: fmt.Sprint("Modifying a VM during a rename process is not allowed"),
```

Comment:
The error message here doesn't seem right to me. This error is returned if user which is not allowed to modify kubevirt components is trying to do so.

Reply:
@danielBelenky WDYT about the new message?